accepted/0000-selective-versions-resolutions.md

```diff
@@ -0,0 +1,495 @@
+
+It is similar to the previous alternative but with a version specification
+in the package designation. This would take this form in the `package.json`:
+```json
+  "devDependencies": {
+    "@angular/cli": "1.0.3",
+    "typescript": "2.2.2",
+    "more dependencies..."
+  },
+  "resolutions": {
+    "typescript@>=2.0.0 <2.3.0": "typescript@2.3.2"
+  }
+```
+
+yarn would then replace matching version specifications with the user's one.
+For example a dependency normally resolved to `typescript@2.2.2` would be
+resolved in practice to `typescript@2.3.2`.
+
+This is too advanced and can be considered a possible extension of this RFC.
+
+## Mapping version specifications as well as packages name
+
+Same as the two above but with a different name on the right-hand side of the
+*resolution*:
+```json
+  "devDependencies": {
+    "@angular/cli": "1.0.3",
+    "typescript": "2.2.2"
+  },
+  "resolutions": {
+    "typescript@>=2.0.0 <2.3.0": "my-typescript-fork@2.3.2"
+  }
+```
+
+or even:
+```json
+  "devDependencies": {
+    "@angular/cli": "1.0.3",
+    "typescript": "2.2.2"
+  },
+  "resolutions": {
+    "typescript": "my-typescript-fork",
+  }
+```
+
+and the version specification would be conserved.
+
+This is too advanced and can be considered a possible extension of this RFC.
+
+# Future extensions
+
+The two alternatives discussed in the section just above, "Mapping version
+specifications" and "Mapping version specifications as well as packages name",
+can be adapted to the current proposition to support these uses cases as well.
+
+# Unresolved questions
+
+## Package designation and single `*`
+
+I am not totally convinced that using a single `*` in a package designation
+is a good idea. Mostly because it introduces much for uncertainty to what it is
+going to be applied. In particular, I'm thinking about defining a *resolution*
+like that and latter adding a dependency which has a dependency that matches
+but is not intended to. I feel like it adds a lot of mental weight for the
+user to manipulate single `*`.
+
+## `--flat` option
+
+I wonder if the `--flat` option of `install` shouldn't be transformed to
+a `flatten` command that would:
+1. Fill in the *resolutions* for all nested dependencies.
+2. Set the `flat` field in the `package.json`.
+
+It makes no real sense to have a flattening mode for `install`:
+1. `install` already follows the `resolutions` field with this RFC.
+2. `install` should be only about building the `node_modules` directory, not
+modifying the the `package.json` IMHO.
+
+If we do that, then the `--flat` option of `add` (and the `flat` option in the
+`package.json`) would apply not to the installation but to the adding,
+upgrading, etc (everything that modify the `package.json`'s dependencies):
+it will ensure that the project stays flattened via the populating of the
+`resolutions` field.
+
+## resolutions in dependencies
+
+If a dependency contains resolutions, should it be taken into account?

```

Comment:
No, same as shrikwraps/lockfiles, every project is responsible for their resolutions.
Also right now this is a Yarn-only feature, we don't really want to get it into the wild yet too much